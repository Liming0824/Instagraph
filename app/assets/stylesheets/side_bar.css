
.for-rainbow .item-box{
  width: 50px;
  height: 50px;
  border-radius: 50%;
  display: flex;
  padding: 1.5px;
  background-color: white;
}

.for-rainbow{
  width: 53px;
  height: 53px;
  border-radius: 50%;
  padding: 2px;
  background-image: linear-gradient(to right,orange,violet);
  align-items: center;
}

.friend-item{
  list-style: none;
  margin: 15px 0 15px 10px;
}

.friend-item .item-box img{
  border-radius: 50%;
  object-fit: cover;
  object-position: 50% 50%;
}

.friend-item{
  margin-left: 5px;
  display: flex;
  flex-direction: row;
}

.info-div{
  margin: auto auto auto 15px;
  display: flex;
  flex-direction: column;
}


.info-div span{
  font-family: 'Ubuntu', sans-serif;
  font-size: 15px;
}
.info-div strong{
  font-family: 'Ubuntu', sans-serif;
  font-size: 10px;
  padding: 5px 0px;
  opacity: 0.5;
}
