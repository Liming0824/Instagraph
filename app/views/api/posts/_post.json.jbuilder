json.extract! post, :id, :image_url
json.photo_image_url url_for(post.photo)
json.posterId post.user_id
json.likes post.likes.map{|like| like.liker_id}
