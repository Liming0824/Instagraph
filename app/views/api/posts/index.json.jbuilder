json.posts do
  @posts.each do |post|
    json.set! post.id do
      json.partial! 'api/posts/post', post: post
    end
  end
end

json.users do
  @users.each do |user|
    json.set! user.id do
      json.extract! user, :id, :username, :user_img_url;
      json.photo_image_url url_for(user.user_photo)
      json.posts user.posts do |post|
          json.partial! 'api/posts/post', post: post
      end
    end
  end
end

#
#
# json.users do
#   @users.each do |user|
#     json.set! user.id do
#       json.extract! user, :id, :username, :user_img_url;
#       json.photo_image_url url_for(user.user_photo)
#       json.posts do
#         user.posts.each do |post|
#           json.set! post.id do
#             json.partial! 'api/posts/post', post: post
#           end
#         end
#       end
#     end
#   end
# end
