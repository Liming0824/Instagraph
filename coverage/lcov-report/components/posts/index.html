<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/posts</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> components/posts
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">19.64% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>11/56</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">7.14% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/14</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">9.38% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/32</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">20.37% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>11/54</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="post_index.jsx"><a href="post_index.jsx.html">post_index.jsx</a></td>
	<td data-value="56.25" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 56%;"></div><div class="cover-empty" style="width:44%;"></div></div></td>
	<td data-value="56.25" class="pct medium">56.25%</td>
	<td data-value="16" class="abs medium">9/16</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="9" class="abs low">3/9</td>
	<td data-value="64.29" class="pct medium">64.29%</td>
	<td data-value="14" class="abs medium">9/14</td>
	</tr>

<tr>
	<td class="file low" data-value="post_item.jsx"><a href="post_item.jsx.html">post_item.jsx</a></td>
	<td data-value="5" class="pic low"><div class="chart"><div class="cover-fill" style="width: 5%;"></div><div class="cover-empty" style="width:95%;"></div></div></td>
	<td data-value="5" class="pct low">5%</td>
	<td data-value="40" class="abs low">2/40</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="12" class="abs low">0/12</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="23" class="abs low">0/23</td>
	<td data-value="5" class="pct low">5%</td>
	<td data-value="40" class="abs low">2/40</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Nov 20 2018 13:11:36 GMT-0500 (EST)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
