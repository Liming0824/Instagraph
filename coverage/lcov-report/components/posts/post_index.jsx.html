<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/posts/post_index.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/posts</a> post_index.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">56.25% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>9/16</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">33.33% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/9</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">64.29% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>9/14</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import { connect } from 'react-redux';
import { fetchPost, fetchPosts, deletePost } from '../../actions/post_actions';
import PostItem from './post_item';
import { FadeLoader } from 'react-spinners';
import { css } from 'react-emotion';
&nbsp;
const override = css`
    display: block;
    margin: 30% auto;
    border-color: red;
`;
&nbsp;
export class PostIndex extends React.Component {
  constructor(props){
    super(props);
    this.props.removeErrors();
  }
&nbsp;
  componentDidMount(){
    this.props.fetchPosts();
  }
&nbsp;
  render(){
    const items = this.props.posts.map(<span class="fstat-no" title="function not covered" >po</span>st =&gt; <span class="cstat-no" title="statement not covered" >&lt;PostItem key={post.id} post={post} deletePost={this.props.deletePost}/&gt;)</span>;
    <span class="missing-if-branch" title="else path not taken" >E</span>if(items.length === 0){
      return (
        &lt;div className='post-item-box'&gt;
          &lt;FadeLoader className={override} sizeUnit={"px"} size={150} color={'grey'} /&gt;
        &lt;/div&gt;
      )
    }else{
<span class="cstat-no" title="statement not covered" >      return (</span>
        &lt;div className='posts-body'&gt;
          {items}
        &lt;/div&gt;
      )
    }
  }
}
&nbsp;
&nbsp;
const mapStateToProps = <span class="fstat-no" title="function not covered" >(s</span>tate) =&gt; {
<span class="cstat-no" title="statement not covered" >  return {</span>
    posts: Object.values(state.entities.posts).reverse(),
    users: Object.values(state.entities.users)
  }
}
&nbsp;
&nbsp;
const mapDispatchToProps = <span class="fstat-no" title="function not covered" >(d</span>ispatch) =&gt; (<span class="cstat-no" title="statement not covered" >{</span>
  fetchPosts: <span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >dispatch(fetchPosts()),</span>
  removeErrors: <span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >dispatch({type: 'REMOVE_ERROR'}),</span>
  deletePost: <span class="fstat-no" title="function not covered" >(i</span>d) =&gt; <span class="cstat-no" title="statement not covered" >dispatch(deletePost(id))</span>
});
&nbsp;
export default connect(mapStateToProps, mapDispatchToProps)(PostIndex);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Nov 20 2018 13:11:36 GMT-0500 (EST)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
