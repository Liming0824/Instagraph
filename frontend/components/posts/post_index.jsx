import React from 'react';
import { connect } from 'react-redux';
import { fetchPost, fetchPosts, deletePost } from '../../actions/post_actions';
import PostItem from './post_item';
import { FadeLoader } from 'react-spinners';
import { css } from 'react-emotion';

const override = css`
    display: block;
    margin: 30% auto;
    border-color: red;
`;

export class PostIndex extends React.Component {
  constructor(props){
    super(props);
    this.props.removeErrors();
  }

  componentDidMount(){
    this.props.fetchPosts();
  }

  render(){
    const items = this.props.posts.map(post => <PostItem key={post.id} post={post} deletePost={this.props.deletePost}/>);
    if(items.length === 0){
      return (
        <div className='post-item-box'>
          <FadeLoader className={override} sizeUnit={"px"} size={150} color={'grey'} />
        </div>
      )
    }else{
      return (
        <div className='posts-body'>
          {items}
        </div>
      )
    }
  }
}


const mapStateToProps = (state) => {
  return {
    posts: Object.values(state.entities.posts).reverse(),
    users: Object.values(state.entities.users)
  }
}


const mapDispatchToProps = (dispatch) => ({
  fetchPosts: () => dispatch(fetchPosts()),
  removeErrors: () => dispatch({type: 'REMOVE_ERROR'}),
  deletePost: (id) => dispatch(deletePost(id))
});

export default connect(mapStateToProps, mapDispatchToProps)(PostIndex);
