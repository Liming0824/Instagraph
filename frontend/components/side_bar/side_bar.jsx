import React from 'react';
import { connect } from 'react-redux';
import {withRouter} from 'react-router-dom';

class SideBar extends React.Component{

  handleClick(){
    this.props.history.push(`/homepage/${this.props.currentUser.username}/${this.props.currentUser.id}`);
  }


  render(){
    return (
      //side bar box should have margin left 30-50px
      <div className='side-bar-box'>
        <div className='current-user-top'>
          <img src={this.props.currentUser.photo_image_url} onClick={this.handleClick.bind(this)}/>
          <span>{this.props.currentUser.username}</span>
        </div>
        <div className='stories-and-watch-all'>
          <span>Stories</span>
          <span>watch all</span>
        </div>
        <div className='friends-stories'>
          //for now just display friends name, pic, after implement, this list should depends on who has new story
          // going to be list of friends, overflow: scroll
        </div>
        <footer>
          <div>
            About Us * Support * Press * Api * Jobs * Privacy *
            Terms * Directory * Profiles * Hashtags * Language
          </div>
          <div>
            @2018 INSTAGRAPH
          </div>
        </footer>
      </div>
    )
  }
}


const mapStateToProps = (state, ownProps) => {
  return {
    currentUser: state.entities.users[state.session.currentUserId]
  }
}

export default withRouter(connect(mapStateToProps)(SideBar));
