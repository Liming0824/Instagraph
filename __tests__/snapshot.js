import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { SideBar } from '../frontend/components/side_bar/side_bar';
import { PostIndex } from '../frontend/components/posts/post_index';


describe('match snapshot', () => {
    it('<SideBar /> snapshot match', () => {
      const props = {
        followings: [],
        currentUser: {},
        searchUsersByIds: jest.fn(),
      };
      const tree = renderer.create(<SideBar {...props}/>).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('<PostIndex /> snapshot match', ()=>{

      const props = {
        removeErrors: jest.fn(),
        fetchPosts: jest.fn(),
        posts: [],
      };
      const tree = renderer.create(<PostIndex {...props}/>).toJSON();
      expect(tree).toMatchSnapshot();
    });


});
