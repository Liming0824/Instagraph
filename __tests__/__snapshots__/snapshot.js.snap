// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot <PostIndex /> snapshot match 1`] = `
<div
  className="post-item-box"
>
  <div
    className="css-1nyi826"
  >
    <div
      className="css-1izzguw"
    />
    <div
      className="css-4bzpi4"
    />
    <div
      className="css-134b1y3"
    />
    <div
      className="css-ht3zcn"
    />
    <div
      className="css-1hu78i0"
    />
    <div
      className="css-jsdrog"
    />
    <div
      className="css-bkcxpe"
    />
    <div
      className="css-1mmzqo5"
    />
  </div>
</div>
`;

exports[`match snapshot <SideBar /> snapshot match 1`] = `
<div
  className="side-bar-box"
>
  <div
    className="current-user-top"
  >
    <img
      onClick={[Function]}
    />
    <span />
  </div>
  <div
    className="stories-and-watch-all"
  >
    <span>
      Stories
    </span>
    <span>
      watch all
    </span>
  </div>
  <div
    className="friends-stories"
  />
  <footer>
    <div>
      @2018 INSTAGRAPH
    </div>
  </footer>
</div>
`;

exports[`should render 0 1`] = `
<div
  className="side-bar-box"
>
  <div
    className="current-user-top"
  >
    <img
      onClick={[Function]}
    />
    <span />
  </div>
  <div
    className="stories-and-watch-all"
  >
    <span>
      Stories
    </span>
    <span>
      watch all
    </span>
  </div>
  <div
    className="friends-stories"
  />
  <footer>
    <div>
      @2018 INSTAGRAPH
    </div>
  </footer>
</div>
`;
